sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridClient.cs

```diff
@@ -151,7 +210,32 @@ public virtual Response PublishCloudEvents(IEnumerable<CloudEvent> events, Cance
 
             try
             {
-                return _serviceRestClient.PublishCloudEventEvents(_hostName, events, cancellationToken);
+                List<CloudEvent> eventsWithSerializedPayloads = new List<CloudEvent>();
+                foreach (CloudEvent cloudEvent in events)
+                {
+                    if (cloudEvent.Data != null)
```

Comment:
nit: just use `event` since the param is named `events`.

Reply:
I think event is a keyword so I couldn't do that

Reply:
ah right 😄